src/main/java/com/rackspace/salus/monitor_management/services/MonitorManagement.java

```diff
@@ -1906,6 +1912,24 @@ private void handleReattachedEnvoy(String tenantId, String resourceId, String en
     return extractEnvoyIds(boundMonitors);
   }
 
+  /**
+   * Removes all bindings associated with the given monitor IDs and resource id.
+   *
+   * @return affected envoy IDs
+   */
+  Set<String> unbindByTenantAndResourceId(String tenantId,
+      String resourceId) {
```

Comment:
My comment above would be easy to address by
```suggestion
  Set<String> unbindByTenantAndResourceId(String tenantId,
      String resourceId, Set<UUID> monitorIdsToUnbind) {
```

and then a stream-filter on the `boundMonitors` retrieved by the findMonitorsBoundToTenantAndResource to include only those in the given set. Could even enhance the JPA query itself to add an `AND IN(:monitorIdsToUnbind)` but I don't feel strongly either way. @GeorgeJahad @harageth any preference to filter via JPA/SQL query or streams?

Reply:
yes, I was thinking to put `AND IN(:monitorIdsToUnbind)` in JPA query but was not aware of that context you explained in above comments.

Reply:
Yeah, let's go with the JPA query change then.